From bd6dbc0fa0037fe8dfaa84549155bd46314bf8ed Mon Sep 17 00:00:00 2001
From: Mark Yao <mark.yao@rock-chips.com>
Date: Wed, 8 Oct 2014 14:38:23 +0800
Subject: [PATCH] CHROMIUM: libdrm: Add header for Mediatek DRM userspace
 interface

Add a header defining the userspace interface exposed by Mediatek DRM
driver.

Old-BUG=chrome-os-partner:38389
Old-BUG=b:26575005
Old-TEST=Use kmstest on oak to verify

Signed-off-by: JB Tsai <jb.tsai@mediatek.com>
Signed-off-by: Jason-JH.Lin <jason-jh.lin@mediatek.com>
(rebased from chromeos-2.4.70)
(stripped down to the header alone)

BUG=b:112720782
Signed-off-by: Kristian H. Kristensen <hoegsberg@chromium.org>

Change-Id: Idcbe76bbc710c4ee5b08af640fd3ccaadf9d33a0
---
 include/drm/mediatek_drm.h | 62 ++++++++++++++++++++++++++++++++++++++
 meson.build                |  2 +-
 2 files changed, 63 insertions(+), 1 deletion(-)
 create mode 100644 include/drm/mediatek_drm.h

diff --git a/include/drm/mediatek_drm.h b/include/drm/mediatek_drm.h
new file mode 100644
index 00000000..51c5dc18
--- /dev/null
+++ b/include/drm/mediatek_drm.h
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2015 MediaTek Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+
+#ifndef _UAPI_MEDIATEK_DRM_H
+#define _UAPI_MEDIATEK_DRM_H
+
+#include "drm.h"
+
+/**
+ * User-desired buffer creation information structure.
+ *
+ * @size: user-desired memory allocation size.
+ *      - this size value would be page-aligned internally.
+ * @flags: user request for setting memory type or cache attributes.
+ * @handle: returned a handle to created gem object.
+ *	- this handle will be set by gem module of kernel side.
+ */
+struct drm_mtk_gem_create {
+	uint64_t size;
+	uint32_t flags;
+	uint32_t handle;
+};
+
+/**
+ * A structure for getting buffer offset.
+ *
+ * @handle: a pointer to gem object created.
+ * @pad: just padding to be 64-bit aligned.
+ * @offset: relatived offset value of the memory region allocated.
+ *	- this value should be set by user.
+ */
+struct drm_mtk_gem_map_off {
+	uint32_t handle;
+	uint32_t pad;
+	uint64_t offset;
+};
+
+#define DRM_MTK_GEM_CREATE              0x00
+#define DRM_MTK_GEM_MAP_OFFSET          0x01
+
+#define DRM_MTK_GEM_CREATE_FLAG_RESTRICTED          (1 << 0)
+#define DRM_MTK_GEM_CREATE_FLAG_ALLOC_SINGLE_PAGES  (1 << 1)
+
+#define DRM_IOCTL_MTK_GEM_CREATE        DRM_IOWR(DRM_COMMAND_BASE + \
+                DRM_MTK_GEM_CREATE, struct drm_mtk_gem_create)
+
+#define DRM_IOCTL_MTK_GEM_MAP_OFFSET    DRM_IOWR(DRM_COMMAND_BASE + \
+                DRM_MTK_GEM_MAP_OFFSET, struct drm_mtk_gem_map_off)
+
+
+#endif /* _UAPI_MEDIATEK_DRM_H */
diff --git a/meson.build b/meson.build
index 3a42b570..708c42d2 100644
--- a/meson.build
+++ b/meson.build
@@ -304,7 +304,7 @@ install_headers('libsync.h', 'xf86drm.h', 'xf86drmMode.h')
 install_headers(
   'include/drm/drm.h', 'include/drm/drm_fourcc.h', 'include/drm/drm_mode.h',
   'include/drm/drm_sarea.h', 'include/drm/i915_drm.h',
-  'include/drm/mach64_drm.h', 'include/drm/mga_drm.h',
+  'include/drm/mach64_drm.h', 'include/drm/mediatek_drm.h', 'include/drm/mga_drm.h',
   'include/drm/msm_drm.h', 'include/drm/nouveau_drm.h',
   'include/drm/qxl_drm.h', 'include/drm/r128_drm.h',
   'include/drm/radeon_drm.h', 'include/drm/amdgpu_drm.h',
-- 
2.48.1.362.g079036d154-goog

